cogeo_mosaic/backend/dynamo.py

```diff
@@ -0,0 +1,81 @@
+import functools
+import itertools
+import os
+from typing import Dict
+
+import mercantile
+
+from cogeo_mosaic.backend.base import BaseBackend
+
+
+class DynamoDBBackend(BaseBackend):
+    """DynamoDB Backend Adapter"""
+
+    def __init__(
+        self, mosaicid: str, region: str = os.getenv("AWS_REGION", "us-east-1")
+    ):
+        self.client = boto3.resource("dynamodb", region_name=region)
+        self.table = self.client.Table(mosaicid)
+        self.mosaic_def = self.fetch_mosaic_definition()
+
+    def metadata(self) -> Dict:
+        return self.mosaic_def
+
+    def tile(self, x: int, y: int, z: int, bucket: str, key: str) -> List[str]:
+        """Retrieve assets for tile."""
+
+        return self.get_assets(x, y, z)
+
+    def point(self, lng: float, lat: float) -> List[str]:
+        """Retrieve assets for point."""
+
+        min_zoom = self.mosaic_def["minzoom"]
+        quadkey_zoom = self.mosaic_def.get("quadkey_zoom", min_zoom)  # 0.0.2
+        tile = mercantile.tile(lng, lat, quadkey_zoom)
+        return self.get_assets(tile.x, tile.y, tile.z)
+
+    @functools.lru_cache(maxsize=512)
+    def fetch_mosaic_definition(self) -> Dict:
+        """Get Mosaic definition info."""
+        # NOTE(kylebarron): may also want to convert Decimal to int/float here
+        return fetch_dynamodb("-1")
```

Comment:
ok so `-1` is a dummy `Key` where we store the a json object with the mosaicJSON metadata right ? 

dumb question, does the Key has to be a number ?

Reply:
Yes, `-1` is the dummy `Key` for storing non-`tiles` metadata.

The Key has to be the same data type throughout. Since quadkeys are strings, all `quadkey` values need to be strings, hence `'-1'`

Reply:
thanks